src/config/server.rs

```diff
@@ -47,3 +48,85 @@ impl Server {
 pub struct Proxy {
   pub url: String,
 }
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct Upstream {
+  #[serde(default = "pool_idle_timeout_default")]
+  pub pool_idle_timeout: u64,
+  #[serde(default = "pool_max_idle_per_host_default")]
+  pub pool_max_idle_per_host: usize,
+  #[serde(default = "keep_alive_interval_default")]
+  pub keep_alive_interval: u64,
+  #[serde(default = "keep_alive_timeout_default")]
+  pub keep_alive_timeout: u64,
+  #[serde(default = "keep_alive_while_idle_default")]
+  pub keep_alive_while_idle: bool,
+  #[serde(default = "proxy_default")]
+  pub proxy: Option<Proxy>,
+  #[serde(default = "connect_timeout_default")]
+  pub connect_timeout: u64,
+  #[serde(default = "timeout_default")]
+  pub timeout: u64,
+  #[serde(default = "tcp_keep_alive_default")]
+  pub tcp_keep_alive: u64,
+  #[serde(default = "user_agent_default")]
+  pub user_agent: String,
+}
+
+impl Default for Upstream {
+  fn default() -> Self {
+    Upstream {
+      pool_idle_timeout: pool_idle_timeout_default(),
+      pool_max_idle_per_host: pool_max_idle_per_host_default(),
+      keep_alive_interval: keep_alive_interval_default(),
+      keep_alive_timeout: keep_alive_timeout_default(),
+      keep_alive_while_idle: keep_alive_while_idle_default(),
+      proxy: proxy_default(),
+      connect_timeout: connect_timeout_default(),
+      timeout: timeout_default(),
+      tcp_keep_alive: tcp_keep_alive_default(),
+      user_agent: user_agent_default(),
+    }
+  }
+}
+
+fn pool_idle_timeout_default() -> u64 {
+  60
+}
+
+fn pool_max_idle_per_host_default() -> usize {
+  200
+}
+
+fn keep_alive_interval_default() -> u64 {
+  60
+}
+
+fn keep_alive_timeout_default() -> u64 {
+  60
+}
+
+fn keep_alive_while_idle_default() -> bool {
+  false
+}
+
+fn proxy_default() -> Option<Proxy> {
+  None
+}
+
```

Comment:
we can omit defining default functions in cases when the default is equal to the type's default as it is for Option and bool.

for attribute it will be just `#[serde(default)]`